django/db/models/base.py

```diff
@@ -1153,7 +1153,8 @@ def _save_table(
                     getattr(self, field.attname) if raw else field.pre_save(self, False)
                 )
                 if hasattr(value, "resolve_expression"):
-                    returning_fields.append(field)
+                    if field not in returning_fields:
```

Comment:
In tandem with #19868 we can collapse this when the else is gone.

Reply:
Nice catch!